### PR TITLE
Add support for phrasing content

### PR DESCRIPTION
**Pull request by wooorm**
Closes GH-56.
Closes GH-57.
Closes GH-58.

### CODE REVIEW

**Comment by wooorm** [comment]
@d4rekanguok here’s how I’d do it. Could you check? The slugs/URLs [aren’t perfect](https://github.com/syntax-tree/mdast-util-toc/pull/58/files#diff-3a6e5559d4d24b8dab8074b558b3974eR127), though. And [they’re different from GitHub](https://github.com/syntax-tree/mdast-util-toc/blob/e2988c8ce0ecd95a7fb3275851b7f17ee85e4f65/test/fixtures/phrasing-content/input.md#html). Although this has always been a problem, so it isn’t new.

**Comment by d4rekanguok** [comment]
I tested it out & it worked great!! Can't wait to see this get merged. Thank you again for the swift response.

File: lib/contents.js [hunk 1]
```diff
@@ -129,6 +131,39 @@ function insert(node, parent, tight) {
   }
 }
 
+function all(children) {
+  var result = []
+  var length = children.length
+  var index = -1
+
+  while (++index < length) {
+    result = result.concat(one(children[index]))
```
Comment (BarryThePenguin) [inline lib/contents.js]: One option could be to only copy `text` nodes

Comment (BarryThePenguin) [reply lib/contents.js]: Hmm... that gets tricky for `image` nodes and the like..

**Comment by wooorm** [comment]
Thanks for reviewing both!

@BarryThePenguin This will definitely break things! Potentially for the better.
I think people don’t typically include syntax in headings, except for a big part of the audience where they use `inline code` to document API functions, which [we’re doing in this project](https://github.com/syntax-tree/mdast-util-toc#api) as well.
I think copying that syntax over is actually better. But that might be an opinion, so we could also make it an option?

**Comment by wooorm** [comment]
@BarryThePenguin What do you think re my previous comment?

**Comment by wooorm** [comment]
@BarryThePenguin Regarding keeping the current behaviour as the default option: I think the new behaviour makes more sense as the default. I propose releasing this as a major, and if issues come in investigate what users would like: 1) the previous behaviour, or 2) some sort of node filtering where they can choose which nodes to keep, etc, or maybe something.

For this PR: I just noticed that [images](https://github.com/syntax-tree/mdast-util-toc/pull/58/files#diff-3a6e5559d4d24b8dab8074b558b3974eR240) are also included in the TOC. Is that a good idea? I don’t think people really use images in headings (except maybe the main heading of a readme, but that doesn’t appear in the TOC), so we can either a) not copy images, or b) wait for what  people want and potentially allow filtering (2 above) of nodes later.

**Comment by BarryThePenguin** [comment]
Sounds good 👍🏻

**Pull request merged**